inch.go

```diff
@@ -552,6 +583,9 @@ func (s *Simulator) runClient(ctx context.Context, ch <-chan []byte) {
 			s.mu.Lock()
 			s.writtenN += s.BatchSize
 			s.mu.Unlock()
+
+			// Update current throughput
+			atomic.AddInt64(&s.latestValues, int64(s.BatchSize))
```

Comment:
Slightly odd to me that there's a mix of mutex locks and atomic math in the Simulator.